Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM docker.io/library/rust:latest as builder
+
+WORKDIR /app
+COPY . .
+RUN cargo build --release
+
+FROM gcr.io/distroless/cc-debian12:nonroot AS runtime
```

Comment:
What is `gcr.io/distroless/cc-debian12:nonroot`? Why not use an official debian image?

Reply:
@casey "Distroless" images contain only an application and its runtime dependencies. They do not contain package managers, shells or any other programs you would expect to find in a standard Linux distribution. See more info @ https://github.com/GoogleContainerTools/distroless

I chose this image because it contains a minimal Linux, glibc runtime for "mostly-statically compiled" languages like Rust and D.

In my opinion the main benefit for `intermodal` is the reduced final image size:
distroless: ~30MB
official debian bookworm slim: ~85MB

If you still prefer to use the official debian image in any case this can be changed to:
`FROM docker.io/library/debian:bookworm-slim AS runtime`